resolwe_bio/processes/workflows/geo_import.py

```diff
@@ -368,17 +368,23 @@ def run(self, inputs, outputs):
                 f"GEO series accessions (GSE) are supported but {inputs.gse_accession} was provided."
             )
 
-        try:
-            gse = GEOparse.get_GEO(geo=inputs.gse_accession, destdir="./")
-        except IOError:
+        gse = None
+        for _ in range(10):
+            try:
+                time.sleep(1)
+                gse = GEOparse.get_GEO(geo=inputs.gse_accession, destdir="./")
+            except IOError:
+                self.warning(f"Download of {inputs.gse_accession} failed. Retrying.")
```

Comment:
Nitpick: IO error is mostly raised when the ID is incorrect or the data is missing. We could skip retrying in that case. Otherwise I would suggest we just handle `IOErrors` as other exceptions.

Reply:
I've simplified a bit and importantly, added a `break` statement. Please take another look.